pkg/ruler/base/ruler.go

```diff
@@ -50,7 +50,7 @@ var (
 
 const (
 	// ringKey is the key under which we store the rulers ring in the KVStore.
-	ringKey = "ring"
+	ringKey = "rulers"
```

Comment:
this needs to be something other than `ring` otherwise the ruler and ingester will conflict if registered under the same prefix.